src/service/url-replacements-impl.js

```diff
@@ -163,6 +164,18 @@ export class UrlReplacements {
       return removeFragment(info.sourceUrl);
     }));
 
+    this.setAsync_('SOURCE_URL', () => {
+      let promise = trackImpressionPromise;
+      if (!promise) {
+        promise = Promise.resolve();
+      }
+      return promise.then(
+        this.getDocInfoValue_.bind(this, info => {
+          return removeFragment(info.sourceUrl);
```

Comment:
I see! I don't want to change this though. it's a one line code, no need to add an extra method for it.